applications/serial_lte_modem/src/slm_at_commands.c

```diff
@@ -519,24 +519,28 @@ int handle_at_clac(enum at_cmd_type cmd_type)
 	return ret;
 }
 
-int slm_at_parse(const char *at_cmd)
+int slm_at_parse(const char *cmd_str, size_t cmd_name_len)
 {
 	int ret = UNKNOWN_AT_COMMAND_RET;
 	int total = ARRAY_SIZE(slm_at_cmd_list);
 
 	for (int i = 0; i < total; i++) {
-		if (slm_util_cmd_casecmp(at_cmd, slm_at_cmd_list[i].string)) {
-			enum at_cmd_type type = at_parser_cmd_type_get(at_cmd);
-
-			at_params_list_clear(&slm_at_param_list);
-			ret = at_parser_params_from_str(at_cmd, NULL, &slm_at_param_list);
-			if (ret) {
-				LOG_ERR("Failed to parse AT command %d", ret);
-				return -EINVAL;
-			}
-			ret = slm_at_cmd_list[i].handler(type);
-			break;
+		const struct slm_at_cmd *const at_cmd = &slm_at_cmd_list[i];
+
+		if (strncmp(cmd_str, at_cmd->string, cmd_name_len)
+		 || at_cmd->string[cmd_name_len] != '\0') {
```

Comment:
If we have a long at command coming and we compare to shorter one, doesn't the at_cmd->string[cmd_name_len] actually point over the allocated memory area?

Reply:
Should you compare the strlens of the commands instead?

Reply:
Ah so it does, please have a comment for this. This is getting bit clever.

Reply:
Not sure which way I prefer it, but the comment clarifies it.